caffe2/CMakeLists.txt

```diff
@@ -256,12 +256,37 @@ endif()
 # ---[ Caffe2 HIP sources.
 if(USE_ROCM)
   # Call again since Caffe2_HIP_INCLUDES is extended with ATen include dirs.
-  IF(BUILD_ATEN)
-     HIP_INCLUDE_DIRECTORIES(${Caffe2_HIP_INCLUDES})
-  ENDIF()
-  IF(BUILD_CAFFE2)
-     set_source_files_properties(${Caffe2_HIP_SRCS} PROPERTIES HIP_SOURCE_PROPERTY_FORMAT 1)
-  ENDIF()
+  if(BUILD_ATEN)
+    get_directory_property(MY_DEFINITIONS COMPILE_DEFINITIONS)
+    if(MY_DEFINITIONS)
+      foreach(_item ${MY_DEFINITIONS})
+        LIST(APPEND HIP_HCC_FLAGS "-D${_item}")
+      endforeach()
+    endif()
+    hip_include_directories(${Caffe2_HIP_INCLUDES})
+  endif()
+
+  # Include all the necessary cuh files.
+  file(GLOB_RECURSE ATen_CUDA_HEADER_SRCS "${CMAKE_CURRENT_LIST_DIR}/../aten/src/*.cuh")
+
+  # Will uncomment the line below after all *.cuh files can compile standalone.
+  # SET(Caffe2_HIP_SRCS ${ATen_CUDA_HEADER_SRCS} ${Caffe2_HIP_SRCS})
+
+  # Necessary in order to build all the source code
```

Comment:
Similarly, this comment is not very useful. Dig deeper. WHY is it necessary in order to build all the source code?